app/Services/User/EmailChange.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Services\User;
+
+use App\Services\BaseService;
+use App\Notifications\ConfirmEmail;
+use App\Exceptions\MissingParameterException;
+
+class EmailChange extends BaseService
+{
+    /**
+     * The structure that the method expects to receive as parameter.
+     *
+     * @var array
+     */
+    private $structure = [
+        'email',
+    ];
+
+    /**
+     * Update email of the user.
+     *
+     * @param array $data
+     */
+    public function execute(array $data)
+    {
+        if (! $this->validateDataStructure($data, $this->structure)) {
+            throw new MissingParameterException('Missing parameters');
+        }
+
+        $user = auth()->user();
```

Comment:
I don't think it's a good idea that you use `auth()->user()` in a service. This assumes that we call the service in the context where we are authenticated which might not be always the case.

Instead, you need to pass the user id as a parameter and then find the user, etc...

More DB queries but at least, the service is not dependant on something else.